lib/quill_delta.dart

```diff
@@ -17,6 +19,11 @@ const _valueEquality = DeepCollectionEquality();
 /// Useful with embedded content.
 typedef DataDecoder = Object Function(Object data);
 
+/// Predicate function
+///
+/// Useful in functional programming with [Delta]
+typedef Predicate<T> = T Function(Operation op);
```

Comment:
I'm a bit concerned about the very generic name of this typedef as it may collide with other libraries. How about going similar route as Dart SDK with inline definition in the method itself?

```dart
Iterable<T> map<T>(T f(E e))
```

Reply:
```dart
Iterable<T> map<T>(T Function(Operation) f)
```
To comply with linter